icu4c/source/i18n/plurrule.cpp

```diff
@@ -905,6 +909,10 @@ AndConstraint::AndConstraint(const AndConstraint& other) {
         if (U_FAILURE(fInternalStatus)) {
             return;
         }
+        if (this->rangeList != nullptr) {
```

Comment:
This doesn't make any sense to me. This code here is in the constructor for `class AndConstraint` and I can't see any code before this line that assigns anything to `this->rangeList` so the code in this if-statement will never be executed. Is that so, or what am I missing?

Reply:
you are right, I didn't realize it is in the constructor code.